src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,168 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> first;
+    private Node<T> last;
+    private int size;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public void add(T value) {
+        Node<T> newNode = new Node<>(null, value, null);
+        if (isEmpty()) {
+            first = newNode;
+            last = newNode;
+            size++;
+            return;
+        }
+        newNode.prev = last;
+        last.next = newNode;
+        last = newNode;
+        size++;
```

Comment:
refactor to the separate method to avoid code duplication
```suggestion
        linkLast(value);
```